### DISCUSSION TITLE
quarto render - ipynb in another folder directory structure to a different dir with project

### DISCUSSION THREAD

**Discussion by bugsbunny88** [quarto-dev/quarto-cli#5290] [Q&A]
Hi , 
I have been able to create a quarto project website.
aim is to export to html , static embedded content that is then in a _site that I just ftp to a server and its served . 
In terms of a workflow I am having a hard time understand how I accomplish this or if its even possible.

Right now I'm a slave of moving ipynb's to the quarto project and then rendering.

What if I was in a totally different directory , and establish a workflow where when I render it renders to that project ?
I've read the docs multiple times, the sections around Projects, the section around virtual environment, and the Rendering section.
I just don't understand enough to know what or how to accomplish this. and could utilize your help.

**Comment by mcanouil** [comment]
Short answer, yes it is possible to render a Quarto project and FTP all `_site` to any server (for some hosts, there is even a `quarto publish` command).

Long answer, why are you moving your notebooks?
A Quarto project is only a directory where all files live, thus if it makes sense to have them all in the same rendered website, then why not put them all in the same directory and define this directory as your project directory?


Following <https://quarto.org/docs/websites/> should get you started with the rendering part.
If indeed and for unknown reasons you need to copy/move files, I suggest you use a "pre-render" script, see <https://quarto.org/docs/projects/scripts.html#pre-and-post-render>.
For the last part, either do the FTP upload by hand and add a "post-render" script which will do the steps for you, again see <https://quarto.org/docs/projects/scripts.html#pre-and-post-render>.

**Reply by bugsbunny88** [reply]
I thought I needed my notebooks in the quarto folder structure when I create a project so it renders it to a target. 
But I believe what you are telling me is I can render, and manually move my html ?
or are you telling me I just need to define the location in _quarto.yml (wherever it is in the folder structure outside of the project) and it will render it to _site by itself ?

**Reply by bugsbunny88** [reply]
as far as scripts go , I'm shocked I was able to get a website up even. so I have no idea how to create a post render script

**Reply by mcanouil** [reply]
The process is described in the links I posted.
"Post-render" script can be R, Python, Bash, TypeScript scripts and will be executed at the end of `quarto render`.
